fip-0026.md

```diff
@@ -598,11 +571,45 @@ eosio_assert((has_auth(SYSTEMACCOUNT) || has_auth(TREASURYACCOUNT) || has_auth(E
 
 Note: I added a `&& from != EscrowContract` and then wrapped the `check()` to see if the caller `!has_auth(EscrowContract)`.
 
-### Added action to existing contract
+#### Contract: fio.address
+#### Modified Action: burnexpired
+This action needs to be updated because there is no expiration to a domain being listed. If a domain is expired and still listed, when it is going to be burned it needs to also be removed as a listing for sale.
+
+#### Added Processing
+- Look up domain that is being burned in the `domainsales` table
+- If found, remove the entry from `domainsales`
+
+```c++
+    for (int i = 0; i < domainburnlist.size(); i++) {
+        const uint128_t burner = domainburnlist[i];
+
+        auto domainsbyname = domains.get_index<"byname"_n>();
+        auto domainsiter = domainsbyname.find(burner);
+
+        if (domainsiter != domainsbyname.end()) {
+            domainsbyname.erase(domainsiter);
+        }
+
+        // ADDED
+        // Find any domains listed for sale on the fio.escrow contract table
+        auto domainsalesbydomain = domainsales.get_index<"bydomain"_n>();
+        auto domainsaleiter = domainsalesbydomain.find(burner)
+        // if found, erase the entry
+        if(domainsaleiter != domainsalesbydomain.end()){
+            domainsalesbydomain.erase(domainsaleiter)
+        }
+        // END ADDED
+    }
+```
+
+#### Added action to existing contract
 
 - `fio.address::xferescrow` has been added to facilitate the transferring of domains between two accounts without the standard fees. This action is only available when called from `fio.escrow`.
```

Comment:
Why should there not be a fee?

Reply:
That was poorly worded. It was not to get around the fees. I had issues using the existing `xferdomain` action early on and it was suggested to create a new action.